### PR TITLE
Scoreboard Update

### PR DESCRIPTION
**Pull request by DeityLink**
- Moved all the End Round information from the chat window to the popup browser. Easier to read, and allows you to check antags and other info such as silicon laws and antag objectives completions while keeping up with the chat.
- Added a few icons to illustrate the End Round popup. Antags are now displayed as they are next to their name.
- Fixed Photography, again, injuries now properly show up on people.
- Harvested Plants are now properly counted and credited for in the End Round popup (and Crew Score)
- Smelted Ores (not just mined) are now properly counted and credited for in the End Round popup (and Crew Score). Whether you smelt them with the Ore Redemption Machine or the Conveyor on the Asteroid doesn't change anything.
- The Round End popup now displays how many rounds of Arena Bomberman occurred during the round, and who won most of them (and how many)

scoreboard_update

### CODE REVIEW

**Comment by ComicIronic** [comment]
What's going on at the bottom of that screenshot?

**Comment by DeityLink** [comment]
I messed up my editing on Gimp (that screenshot is a compilation of popups from multiple rounds)

File: code/game/gamemodes/blob/blob_finish.dm [hunk 1]
```diff
@@ -13,40 +13,43 @@
...
 <B>The alien organism has been eradicated from the station</B>"}
 
 		var/datum/station_state/end_state = new /datum/station_state()
 		end_state.count()
 		var/percent = round( 100.0 *  start_state.score(end_state), 0.1)
-		world << "<B>The station is [percent]% intact.</B>"
+		completion_text += "<br><B>The station is [percent]% intact.</B>"
 		log_game("Blob mode was won with station [percent]% intact.")
-		world << "<span class='notice'>Rebooting in 30s</span>"
+		completion_text += "<br><span class='notice'>Rebooting in 30s</span>"
```
Comment (ComicIronic) [inline code/game/gamemodes/blob/blob_finish.dm]: I'd suggest keeping this as a world notice.

Comment (DeityLink) [inline code/game/gamemodes/blob/blob_finish.dm]: oh snap, you're right

**Comment by DeityLink** [comment]
BUT IT'S NOOOOOOOOOOOOOOOOOOOOOOOOT ;_;

**Comment by BestName** [comment]
How will I read the end round info from my logs now?
The pop up windows aren't logged.

**Comment by clusterfack** [comment]
He can use log_game for all the info to be sent to logs

**Comment by DeityLink** [comment]
Alright then (though the logs will be filled with  )

**Comment by d3athrow** [comment]
Just seems to be moving the gamemode end stuff to a window so you dont have to dick with the scrolling bug. also fixes some other bugs as well. so i'll pull it.

File: code/game/gamemodes/vampire/vampire.dm [hunk 2]
```diff
@@ -141,15 +148,21 @@
 				text += "<br><font color='red'><B>The [special_role_text] has failed!</B></font>"
 				feedback_add_details("traitor_success","FAIL")
 
-		world << text
-	return 1
+	return text
```
Comment (tigercat2000) [inline code/game/gamemodes/vampire/vampire.dm]: Reading over this- vampire auto_declare_completion is inconsistent with.. pretty much every other gamemode.

Comment (DeityLink) [inline code/game/gamemodes/vampire/vampire.dm]: how exactly?

Comment (DeityLink) [inline code/game/gamemodes/vampire/vampire.dm]: OH

Comment (DeityLink) [inline code/game/gamemodes/vampire/vampire.dm]: Wait no, I misread, well then what are you even talking about?

Comment (tigercat2000) [inline code/game/gamemodes/vampire/vampire.dm]: Dead vamps don't have their icon's .Turn('ed, and pretty much every other gamemode has text += "<br><hr>" before returning the text.

Comment (DeityLink) [inline code/game/gamemodes/vampire/vampire.dm]: well you've got a good eye, or you saw something fishy on the scoreboard at a round end. I'll make a PR to fix that in a moment.

**Pull request merged**